python_modules/dagster/dagster/_core/snap/snap_to_yaml.py

```diff
@@ -23,12 +23,35 @@ def _filter_empty_dicts(to_filter: Any) -> Any:
         return {k: v for k, v in filtered_dict.items() if v is not None and v != {}}
 
 
+def _cleanup_run_config_dict(run_config_dict: Any) -> Any:
+    """Performs cleanup of the run config dict to remove empty dicts and strip the default executor
+    config if it has not been overridden, to make the output more readable.
+    """
+    # We manually strip the default executor config because
+    # the max_concurrent default value is a sentinel and can be confusing
+    # if presented to the user:
+    # execution:
+    #   config:
+    #     multiprocess:
+    #       max_concurrent: 0
+    if (
+        run_config_dict.get("execution", {})
+        .get("config", {})
+        .get("multiprocess", {})
+        .get("max_concurrent", None)
+        == 0
```

Comment:
this is p gross - did you consider changing the schema so that it handles unset instead of default value 0 ?

Reply:
#15389